### PR TITLE
4760 improve vendor api versioning docs

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context

https://trello.com/c/TaSS53A7/4760-improve-documentation-for-adding-a-new-version

## Changes proposed in this pull request
Big update to the VendorAPI docs. 

We've tried to structure it in a more sequential style, so each section implements an incremental change to introducing a minor version, which ends up being released into prod. 

There is some vagueness around adding documentation, as we haven't really got a consistent approach yet. 

We've also skipped out on documenting the changes that we've been making to the `meta` sections of API responses, as they're kind of hard to document consistently. They also pretty much follow the same format as presenter updates, so it's not like they're completely different from what's there. Let us know what you think though!

## Guidance to review
Check out `docs/versioning_vendor_api.md` to see the changes.

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
👍 💯 , a few very minor comments

File: docs/versioning_vendor_api.md [hunk 1]
```diff
@@ -1,301 +1,451 @@
...
 
-## Introducing new changes
+Minor version updates are only for non-breaking changes. A key design principle is maintaining backwards compatibility with all prior minor versions.
 
-Changes in the API are specified in `VersionChange` classes and each version can be made up of one or more classes. Every change class must be configured in `app/lib/vendor_api.rb`
+When incrementing a minor version, the following changes can be introduced:
+ - Adding fields to an existing schema
+ - Adding a new action to an existing controller
+ - Adding a new resource
+ - Adding metadata fields
```
Comment (cwrw) [inline docs/versioning_vendor_api.md:15]: we don't have a section to explain metadata, do we want to add one?

Comment (cwrw) [reply docs/versioning_vendor_api.md:15]: spoke in person:
1) Worth adding a line for the metadata section
2) Add an explanation for v1.1 spec and adding to it incrementally vs the draft

File: docs/versioning_vendor_api.md [hunk 2]
```diff
@@ -1,301 +1,451 @@
...
-## Documenting changes
+Update the `config/vendor_api/v1.2.yml` and the `config/vendor_api/draft.yml`
 
-Changes to the API are documented in the release_notes file, after specifying the version and date that the version was released.
+## Releasing a new version to Sandbox
 
-For a minor version release
+In `app/lib/vendor_api.rb` update the `VERSION` so that it is greater than or equal to the version that you want to expose in the sandbox environment.
 
-```markdown
```
Comment (cwrw) [inline docs/versioning_vendor_api.md:281]: spoke in person: can we add the release notes back

File: docs/versioning_vendor_api.md (hunk 1 above)
Comment (SamJamCul) [reply docs/versioning_vendor_api.md:15]: Changes made - the docs bit has been expanded on, and there's a section that sets out the metadata stuff in some detail

**Pull request merged**